### PR TITLE
feat: automated release script

### PR DESCRIPTION
**Pull request by chris-hoefgen**
- add check for conventional commit PR titles
- updated automated release to work like our other sdks and packages, using new prerelease and release actions

### CODE REVIEW

**Review by JamieSinn — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -49,3 +70,18 @@ jobs:
...
+          tag: ${{ steps.prepare-release.outputs.next-release-tag }}
+          target: main
+          prerelease: ${{ github.event.inputs.prerelease }}
+          draft: ${{ github.event.inputs.draft }}
+          changelog: ${{ steps.prepare-release.outputs.changelog }}
+
+      - name: Display link to release
+        run: |
+          echo "::notice title=Release ID::${{ steps.create-release.outputs.release-id }}"
+          echo "::notice title=Release URL::${{ steps.create-release.outputs.release-url }}"
```
[the remark this answers is not in the document]
Comment (chris-hoefgen) [reply .github/workflows/release.yml:87]: ok, so goreleaser handles the actual GitHub release part of things?

[the remark this answers is not in the document]
Comment (JamieSinn) [reply .github/workflows/release.yml:87]: Yep

[the remark this answers is not in the document]
Comment (chris-hoefgen) [reply .github/workflows/release.yml:87]: cool, i'll just remove that part then as you suggested

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,36 +1,56 @@
...
+        type: boolean
+      version-increment-type:
+        description: 'Which part of the version to increment:'
+        required: true
+        type: choice
+        options:
+          - major
+          - minor
+          - patch
+        default: 'patch'
```
Comment (JamieSinn) [inline .github/workflows/release.yml:25]: Given we don't actually use these values - we should probably remove them

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -1,36 +1,56 @@
...
           fetch-depth: 0
-      - uses: actions/setup-go@v3
+
+      - uses: DevCycleHQ/release-action/prepare-release@main
+        id: prepare-release
+        with:
+          github-token: ${{ secrets.AUTOMATION_USER_TOKEN }}
+          prerelease: ${{ github.event.inputs.prerelease }}
+          draft: ${{ github.event.inputs.draft }}
+          version-increment-type: ${{ github.event.inputs.version-increment-type }}
```
Comment (JamieSinn) [inline .github/workflows/release.yml:47]: Same here - I'd just remove this

File: .github/workflows/release.yml (hunk 2 above)
Comment (chris-hoefgen) [reply .github/workflows/release.yml:25]: done

**Comment by chris-hoefgen** [comment]
Ok, so after the feedback the base release yml didn't change much, just switched to being manual workflow instead of on version creation.  Given that we could just rollback the whole thing if you don't think it is worth it @JamieSinn

**Comment by JamieSinn** [comment]
I think we should have consistent release strategies, so I'd keep the base change of making it manual via dispatch.

**Comment by chris-hoefgen** [comment]
sounds good.  In which case I think i have resolved your issues.

**Pull request merged**